Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -658,47 +632,43 @@ this has no time dependence prior to this operation.
  * ignored, i.e., no time index is added.
 **/
 void ComponentInfo::merge(const ComponentInfo &other) {
-  checkNoTimeDependence();
-  const auto &toMerge = buildMergeIndicesSync(other);
-  for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals->size();
+  const auto &toMerge = buildMergeIndices(other);
+  // Merging the detectorInfo has to be done before we update scanIntervals
+  m_detectorInfo->merge(*other.m_detectorInfo, toMerge);
+  for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals.size();
        ++timeIndex) {
     if (!toMerge[timeIndex])
       continue;
-    auto &scanIntervals = m_scanIntervals.access();
     auto &positions = m_positions.access();
     auto &rotations = m_rotations.access();
-    scanIntervals.push_back((*other.m_scanIntervals)[timeIndex]);
+    m_scanIntervals.push_back(other.m_scanIntervals[timeIndex]);
     const size_t indexStart = other.linearIndex({0, timeIndex});
     size_t indexEnd = indexStart + nonDetectorSize();
     positions.insert(positions.end(), other.m_positions->begin() + indexStart,
                      other.m_positions->begin() + indexEnd);
     rotations.insert(rotations.end(), other.m_rotations->begin() + indexStart,
                      other.m_rotations->begin() + indexEnd);
-    ++m_scanCounts;
   }
-
-  m_detectorInfo->merge(*other.m_detectorInfo);
 }
 
 std::vector<bool>
-ComponentInfo::buildMergeIndicesSync(const ComponentInfo &other) const {
+ComponentInfo::buildMergeIndices(const ComponentInfo &other) const {
   checkSizes(other);
-  std::vector<bool> merge(other.m_scanIntervals->size(), true);
-
-  for (size_t t1 = 0; t1 < other.m_scanIntervals->size(); ++t1) {
-    for (size_t t2 = 0; t2 < m_scanIntervals->size(); ++t2) {
-      const auto &interval1 = (*other.m_scanIntervals)[t1];
-      const auto &interval2 = (*m_scanIntervals)[t2];
+  std::vector<bool> merge(other.m_scanIntervals.size(), true);
+  for (size_t t1 = 0; t1 < other.m_scanIntervals.size(); ++t1) {
+    for (size_t t2 = 0; t2 < m_scanIntervals.size(); ++t2) {
+      const auto interval1 = other.m_scanIntervals[t1];
+      const auto interval2 = m_scanIntervals[t2];
       if (interval1 == interval2) {
         for (size_t compIndex = 0; compIndex < nonDetectorSize(); ++compIndex) {
```

Comment:
This does not look correct. Don't you have to use `size()` here to check all components including detectors?